engine/src/cmd/collide2/Ice/IceMemoryMacros.h

```diff
@@ -85,14 +91,14 @@
 	//!	\see		FillMemory
 	//!	\see		StoreDwords
 	//!	\see		CopyMemory
-	inline_ void MoveMemory(void* dest, const void* src, udword size)	{ memmove(dest, src, size);		}
+	inline_ void MoveMemory(void* dest, const void* src, size_t size)	{ memmove(dest, src, size);		}
 
 	#define SIZEOFOBJECT		sizeof(*this)									//!< Gives the size of current object. Avoid some mistakes (e.g. "sizeof(this)").
 	//#define CLEAROBJECT		{ memset(this, 0, SIZEOFOBJECT);	}			//!< Clears current object. Laziness is my business. HANDLE WITH CARE.
-	#define DELETESINGLE(x)		{ delete x; x = null; }		//!< Deletes an instance of a class.
-	#define DELETEARRAY(x)		{ delete []x; x = null; }		//!< Deletes an array.
-	#define SAFE_RELEASE(x)		if (x) { (x)->Release();		(x) = null; }	//!< Safe D3D-style release
-	#define SAFE_DESTRUCT(x)	if (x) { (x)->SelfDestruct();	(x) = null; }	//!< Safe ICE-style release
+	#define DELETESINGLE(x)		{ delete x; x = nullptr; }		//!< Deletes an instance of a class.
+	#define DELETEARRAY(x)		{ delete []x; x = nullptr; }		//!< Deletes an array.
+	#define SAFE_RELEASE(x)		if (x) { (x)->Release();		(x) = nullptr; }	//!< Safe D3D-style release
+	#define SAFE_DESTRUCT(x)	if (x) { (x)->SelfDestruct();	(x) = nullptr; }	//!< Safe ICE-style release
```

Comment:
can we turn these into inline functions?

Reply:
@BenjamenMeyer Can you give an example of what one of these would look like as an inline function?

Reply:
@stephengtuggy 
```C++
inline void DELETESINGLE(void** x) { delete *x; *x = nullptr; }
inline void SAFE_RELEASE(void** x) { if (*x) { (x)->Release(); (x) = nullptr; }
```
Or better yet (one of the few uses of a template):
```C++
template<typename T>
void DELETESINGLE(T x) {
    delete x;
    x = nullptr;
}
template<typename T>
void SAFE_RELEASE(T x) {
    if (x) {
        (x)->Release();
        (x) = nullptr;
    }
}
```

Or something like that...haven't tested the code but that should be about right.

Reply:
I'd also consider renaming to lower case as this would no longer be a macro.

Reply:
Converted to inline functions (template versions).

Reply:
Thanks! Looks great!